### ISSUE TITLE
URL usage not bound to file system instance

### ISSUE THREAD

**Issue by janwillemkl** [appliedAI-Initiative/lakefs-spec#145]
Consider the following example:

```python
import pandas as pd
from lakefs_spec import LakeFSFileSystem

fs = LakeFSFileSystem(
    host="http://localhost:8000",
    username="<USERNAME>",
    password="<PASSWORD>",
)

df = pd.DataFrame.from_dict({"a": [0, 1, 2]})

with fs.transaction as tx:
    df.to_csv("lakefs://example/main/data.csv")
    tx.commit("example", "main", "Some data")
```

I would expect the `df.to_csv` to use the `LakeFSFileSystem` that I have defined above. However, this code fails with a `urllib3.exceptions.LocationValueError: No host specified.` error.

If I set the `LAKEFS_HOST`, `LAKEFS_USERNAME`, and `LAKEFS_PASSWORD` environment variables, it works. (Probably the same applies when using the configuration in the YAML file)

**Version: 0.3.0**

**Comment by janwillemkl** [comment]
You would have to use it as follows to make it part of the transaction.

```python
with fs.open("lakefs://example/main/data3.csv") as f:
            df.to_csv(f)
```

**Comment by janwillemkl** [comment]
However, in the above example you then also have to be careful not to pass the `storage_options` anymore when switching to zero-config.

This is something we have to document very carefully.

**Comment by AdrianoKF** [comment]
Just for reference, [this](https://github.com/pandas-dev/pandas/blob/55d78caa38925e8cf94623adcd0721cc15a56bdd/pandas/io/common.py#L401-L454) is where Pandas calls `fsspec.open` for accessing files through `fsspec` (this might also be interesting for the HDF5 issue).

**Comment by AdrianoKF** [comment]
Will be addressed in documentation through #151
